src/pint/models/timing_model.py

```diff
@@ -422,32 +512,25 @@ def replicate(self, components=[], copy_component=False):
         new_tm.top_level_params = self.top_level_params
         return new_tm
 
-    def map_component(self, component):
-        comps = self.components
-        if isinstance(component, str):
-            if component not in list(comps.keys()):
-                raise AttributeError("No '%s' in the timing model." % component)
-            comp = comps[component]
-        else:  # When component is an component instance.
-            if component not in list(comps.values()):
-                raise AttributeError(
-                    "No '%s' in the timing model." % component.__class__.__name__
-                )
-            else:
-                comp = component
-        comp_type = self.get_component_type(comp)
-        comp_type_list = getattr(self, comp_type + "_list")
-        order = comp_type_list.index(comp)
-        return comp, order, comp_type_list, comp_type
-
     def get_component_of_category(self):
         category = defaultdict(list)
         for cp in self.components.values():
             category[cp.category].append(cp)
         return dict(category)
 
-    def add_param_from_top(self, param, target_component):
-        """Add a parameter to a timing model component."""
+    def add_param_from_top(self, param, target_component, setup=False):
```

Comment:
Weren't we going to get rid of this since we have to specify a target_component no matter what?